src/tiled/commanddialog.cpp

```diff
@@ -76,17 +83,56 @@ void CommandDialog::setShortcut(const QKeySequence &keySequence)
         mUi->treeView->model()->setShortcut(current, keySequence);
 }
 
-void CommandDialog::updateKeySequenceEdit(const QModelIndex &current, const QModelIndex &)
+void CommandDialog::setSaveBeforeExecute(int state)
+{
+    const QModelIndex &current = mUi->treeView->currentIndex();
+    if (current.row() < mUi->treeView->model()->rowCount(QModelIndex()))
```

Comment:
Please add `QModelIndex()` as the default parameter of the `parent` for `rowCount`. It is the default in the base class already, and then we can avoid being needlessly verbose here.